torch/_inductor/fx_passes/mkldnn_fusion.py

```diff
@@ -253,6 +253,69 @@ def fn(match, *args, **kwargs):
 
         return fn
 
+    def _is_valid_computation_mlp_silu_mul_fusion():
+        def fn(match):
+            computation_nodes = filter_nodes(
+                match.nodes, mkldnn._linear_pointwise.default
+            )
+            if len(computation_nodes) != 2:
+                return False
+            # Check for same Activation
+            act = computation_nodes[0].args[0]
+            if computation_nodes[1].args[0] != act:
+                return False
+            # Check for different wgt
+            wgt = computation_nodes[0].args[1]
+            if computation_nodes[1].args[1] == wgt:
+                return False
+            if (
+                wgt.meta.get("val").dtype != torch.bfloat16  # type: ignore[union-attr]
+                or computation_nodes[0].args[1].meta.get("val").dtype != torch.bfloat16  # type: ignore[union-attr]
+            ):
+                # TODO: Extending to support dtype other than bfloat16
+                return False
+            wgt_size = computation_nodes[0].args[1].meta.get("val").size()  # type: ignore[union-attr]
+            # Check for same weight size
+            if computation_nodes[1].args[1].meta.get("val").size() != wgt_size:  # type: ignore[union-attr]
+                return False
+            # Ensure max autotune used with CPP backend
+            if not (
+                torch._inductor.config.max_autotune
+                and "CPP" in torch._inductor.config.max_autotune_gemm_backends
+            ):
+                return False
+
+            return True
+
+        return fn
+
+    def _register_mlp_silu_mul_lowering(pattern):
+        @register_lowering_pattern(
+            pattern,
+            extra_check=_is_valid_computation_mlp_silu_mul_fusion(),
+            pass_number=0,
+        )
+        def fn(match, *args, **kwargs):
+            # TODO: Should take it as a broad linear-unary?
```

Comment:
I have the same question. Why do we treat it as a "linear unary" fusion pattern? Should we treat it as an MLP pattern by its own? Can we do things like this - we make sure the pattern matching happens and can be lowered to the template codegen in the `extra_check` function and do nothing if the template codegen is not applicable?

Reply:
Thanks for the comment and make the corresponding changes. We will do all the checks inside `extra_check` function when do pattern matches.